src/index.ts

```diff
@@ -0,0 +1,89 @@
+import { RefObject, useEffect, useRef, useState } from 'react';
+
+export interface Options {
+  accept?: string;
+  multiple?: boolean;
+}
+
+export interface OnChange<T extends Event = Event> {
+  (event: T): void;
+}
+
+export type Return = {
+  ref: RefObject<HTMLInputElement>;
+  files: File[] | null;
+};
+
+function useInputFile(
+  args: {
+    ref?: RefObject<HTMLInputElement>;
+    options?: Options | undefined;
+    onChange?: OnChange;
+  } = {}
+): Return {
+  const [files, setFiles] = useState<File[] | null>(null);
+  const defaultRef = useRef<HTMLInputElement>(null);
```

Comment:
why do you need this?

Reply:
If user didn't provide `ref`, I want to provide ref by hook.